version: '3.8'

services:
  db:
    image: postgres:12.4-alpine
    volumes:
      - ./postgres_data:/var/lib/postgresql/data
    environment:
      POSTGRES_USER: skymarket
      POSTGRES_PASSWORD: skymarket
      POSTGRES_DB: skymarket
    restart: always
    ports:
      - "5432:5432"
    networks:
      - db_network
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U skymarket -d skymarket" ]
      interval: 5s
      timeout: 5s
      retries: 5


  migrations:
    build: ./backend_django/.
    depends_on:
      db:
        condition: service_healthy
    command: >
      sh -c "python manage.py migrate &&
             python manage.py loadall"
    networks:
      - db_network


  api:
    build: ./backend_django/.
    image: backend_django_migrations:latest
    depends_on:
      migrations:
        condition: service_completed_successfully
    networks:
      - nginx_network
      - db_network


  frontend:
    build: ./frontend_react/.
    volumes:
      - frontend:/app/build
    depends_on:
      - db
      - api
    networks:
      - nginx_network


  nginx:
    image: nginx:1.19.3
    ports:
      - "8000:8000"
      - "3000:3000"
    volumes:
      - ./backend_django/django_static/:/usr/share/nginx/html/django_static/
      - ./backend_django/django_media/:/usr/share/nginx/html/django_media/
      - ./deploy/nginx.conf:/etc/nginx/conf.d/default.conf
      - frontend:/usr/share/nginx/html/
    restart: always
    depends_on:
      - db
      - frontend
    networks:
      - nginx_network


volumes:
  postgres_data:
  frontend:


networks:
  nginx_network:
    driver: bridge
  db_network:
    driver: bridge
